python-checks/src/main/java/org/sonar/python/checks/IgnoredSystemExitCheck.java

```diff
@@ -154,7 +165,14 @@ private static boolean handleCaughtException(SubscriptionContext ctx, Expression
     }
 
     if (SYSTEM_EXIT_EXCEPTION_NAME.equals(caughtExceptionName)) {
-      ctx.addIssue(caughtException, MESSAGE_NOT_RERAISED_CAUGHT_EXCEPTION);
+      var issue = (IssueWithQuickFix) ctx.addIssue(caughtException, MESSAGE_NOT_RERAISED_CAUGHT_EXCEPTION);
+      if (visitor.passStatement != null) {
```

Comment:
I think this condition is not robust enough. We could fail to suggest a quick fixes in some cases where we probably should be able to provide one easily. Also, this can break the code in some (admittedly contrived) cases. For example, the following snippet:

```
try:
  foo()
except SystemExit:
  pass
  x = 42
  print("system exit caught")
  ...
```
would be remplaced with:
```
try:
  foo()
except SystemExit:
  raise
  x = 42
  print("system exit caught")
  ...
```
and would change execution.

Reply:
done